lib/packwerk/const_node_inspector.rb

```diff
@@ -10,13 +10,11 @@ class ConstNodeInspector
 
     def constant_name_from_node(node, ancestors:)
       return nil unless Node.constant?(node)
-
       # Only process the root `const` node for namespaced constant references. For example, in the
       # reference `Spam::Eggs::Thing`, we only process the const node associated with `Spam`.
-      parent = ancestors.first
-      return nil if parent && Node.constant?(parent)
+      return nil if root_constant?(ancestors)
```

Comment:
The name `root_constant?` is the opposite of what this check is doing — as the comment says, we want to ignore every node _except_ `Spam` in `Spam::Eggs::Thing`, so we return `nil` if we find ourselves trying to process a node like `Eggs` or `Thing` whose parent is another constant. If you want to keep the name `root_constant?` then the conditional should be `unless root_constant?` and the behaviour of the predicate should be reversed.